rpc/write.go

```diff
@@ -14,6 +24,39 @@ func (sc *Client) Declare(context.Context, types.Transaction) (*types.AddTxRespo
 	panic("'starknet_addDeclareTransaction' not implemented")
 }
 
-func (sc *Client) Deploy(context.Context, types.Transaction) (*types.AddTxResponse, error) {
-	panic("'starknet_addDeployTransaction' not implemented")
+// AddDeployTransaction allows to declare a class and instantiate the
+// associated contract in one command. This function will be deprecated and
+// replaced by AddDeclareTransaction to declare a class, followed by
+// AddInvokeTransaction to instantiate the contract. For now, it remains the only
+// way to deploy an account without being charged for it.
+func (sc *Client) AddDeployTransaction(ctx context.Context, contractAddressSalt string, constructorCallData []string, contractDefinition types.ContractClass) (*AddDeployTransactionOutput, error) {
+	program, ok := contractDefinition.Program.(string)
+	if !ok {
+		data, err := json.Marshal(contractDefinition.Program)
+		if err != nil {
+			return nil, err
+		}
+		program, err = encodeProgram(data)
+		if err != nil {
+			return nil, err
+		}
+	}
+	contractDefinition.Program = program
+
+	var result AddDeployTransactionOutput
+	err := sc.do(ctx, "starknet_addDeployTransaction", &result, contractAddressSalt, constructorCallData, contractDefinition)
+
```

Comment:
nit: prefer to be explicit here with the err return
```suggestion
	if err := sc.do(ctx, "starknet_addDeployTransaction", &result, contractAddressSalt, constructorCallData, contractDefinition); err != nil {
		return nil, err
	}

```